src/mappings.jl

```diff
@@ -69,3 +72,56 @@ end
 
 f01(x) = 1 .- exp.(.- x.^2)
 f01_inv(y) = sqrt.(.- log.(1 .- y))
+
+"""
+    Piecewise_positive()
+
+Returns a function and a inverse function inverse function
+to map numbers to larger than 0. 
+
+"""
+function Piecewise_positive()
+    return f02, f02_inv
+end
+
+f02(x) = ifelse.(x .> 0, one(eltype(x)) .+ x,one(eltype(x))./(one(eltype(x)).-x))
+f02_grad(x) = ifelse.(x .> 0, one(eltype(x)) , one(eltype(x))./abs2.(one(eltype(x)).-x))
+f02_inv(y) = ifelse.(y .> 1, y .- one(eltype(y)), one(eltype(y)) .- one(eltype(y))./y)
+
+function ChainRulesCore.rrule(::typeof(f02), x)
+    Y = f02(x)
+    function aux_pullback(barx)
+        return zero(eltype(Y)), barx .* f02_grad(x)
+    end
+    return Y, aux_pullback
+end
+
+
+"""
+    Pow4_positive()
+
+Returns a function and a inverse function inverse function
+to map numbers to larger than 0. 
+
+"""
+function Pow4_positive()
+    return f03, f03_inv
+end
+
+f03(x) = abs2.(abs2.(x))
+f03_inv(y) = sqrt.(sqrt.(y))
+
+
+"""
+    Abs_positive()
+
+Returns a function and a inverse function inverse function
+to map numbers to larger than 0. 
+
+"""
+function Abs_positive()
+    return f04, f04_inv
+end
+
+f04(x) = abs.(x)
+f04_inv(y) = abs.(y)
```

Comment:
strictly speaking `y` instead of `abs.(y)` would be better

Reply:
fine with me. since y is supposed to be always positive it does not matter